### PR TITLE
Topology webhook config

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind bug
/platform azure

**What this PR does / why we need it**:
Fixes a bug introduced with https://github.com/gardener/gardener-extension-provider-azure/pull/631 on non-Azure seeds.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Fix a bug introduced with https://github.com/gardener/gardener-extension-provider-azure/pull/631 on non-Azure seeds. The topology webhook will now rely on the managed-resource's injected information to get seed's information.
```

### CODE REVIEW

File: example/controller-registration.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ metadata:
...
...fSDnIp+HDtQrkC0esdjcJhBOGwu42Auuqe+5lvDoaCthS+Xj++HNG34Y4/h9RAPRZ35WRod04sPjLd0xYLFWcbpDuobE2YWWCiolsJqIxkAh9ey3BJWTs7BlpjFbetny8uXZedMSYhv5v47obbb/2uR/tFN5/3c87N7/vJ9UOuaDQwyL2JnP3JKg96tOY5D2vhT3N9JbPA3dA+0t3vz1KGh2Q+NBGes1eCulYIb0FPPy6+jkoUThfFe3tOkPInrpu0fqzqelFxygJjADqZ0oa3grqIqLnVdgQ7n694LaixkHypZsGcarWyEgit4GB1mSo1FYbYmFij5NXBdTivmVuNK2818AINS3OXwip6I2EXUT2M7hukeHmlO7/pd33d1G8cvUov939nfL/t/h3qjz/9xLEsGUXFNGYeKlXKuwzJ47/HJOzR7AFLgPpDMGuRN1UAmFTOl8kl+yGBkxmCdXp2E6FXds9XrmQSkMjy4fruI5MuB4b/gtFlD7f3iVBIg8Yi7nK32GuU7nVzU41997w2WvqFD3d195rWqnj3Ncv9czol8m5Pvh98NefuSeZ4x6PSPkBysyD6Ho6cgMiwKkNBiPo1wLleRHqGshzPMm68CM4ME1YPpUyhoYI3RvPZQ4+zAh0qCvRtJNyM4QvxZPuKypVJxxWQPQGOmmHAm9aiTWhPzr371SXBXPK3kvdRVfk7qz0vhc5ddERiUDJ3xtnJqOaJaI2DBuffBvYNDzes8MsZx76SKbwQJhOciPj5t/zvxwNlhS9F8O+A27A1714CgEdosxElbWbQq7kvQwnPvsMo/KFmUtunT3d2UxLtf9HXvYlxn67t6RPRrZ7//cvRpVeiXOwY7G4oNt271ewfnK1125J3dCdnd3ZJYKeRkNx3vDXq8Y6T0xtBQGe0/U6aKeo+utfzq07uFQvMIKtCgCDX5N5IVixouaDRD8ec2RjN+Xb1+OdvBn/hJl6R1Kp3oRgFh3W1f5TWz5W5M7Oy88mVlYLRdjH8SZKZCwdxK2Znn8lnrp8zB+7gGW0PkEvse9nsJh0jPHrM9HQCl6NTfJS6GlspkrUBjYyn4nCqdl7OzJvcARZ4J8Uy+JgkKe2syTu+Sj+lrGNbWMm2sZd9Zql7rUpS51qUtd6lKXutSlLnWpS13qUpe61KUudalLXepSl7rUpS51aZP0//9NgUUAyAAA
```
Comment (MartinWeindel) [inline example/controller-registration.yaml:8]: I have a similar problem with shoot-dns-service and shoot-cert-service, that the generated charts are different on Mac and Linux. So here I inserted the generated charts with Linux.

```suggestion
  chart: H4sIAAAAAAAAA+w9a3MbN5L5zF+BYpKK7fMMH3plueW7oyXFUcWWWaLiVGqzpwJnIHLi4cxkHpKZeP/7deNFzItD0jLtJERl1yLQABqN7kajgcZMaeyygMUWe5eyIPHCwIri8M5zIYv+nsWs88UHpy6kk6Mj/i+k4r/8797BYa9/1D8+xvze8cFB9wty9OFdN6csSWlMyBdxGKar4JrK/6Rp2jj/pzMap/aCzv1t+8AJPj48rJ3//lFh/vuQ1f+CdB9yoHXpbz7/NPLesBjnfUDuei0aRcZP++DQ7louu2u5LHFiL0p5wZB8z/w5cZAxyG0Yk3TGyAvJSGSIXENGkomI5qtWQOdsQBoZrnVXhcCnptNfNTXLvxs69jT8kD4a5L/f7/cK+v/kELL28r+D1OmQ0zBaxN50lpJHzmPS7/b+QcbDERmfExBtGvAf9PbW8z2aMuKE84gGC5sMfZ/wagmJWcLiO+ba5HrmJQRAGYF/fc8BlmIuyQLUBKglhhF14J9xeJveU1ATLwXIU3Jnkz7oCodFKaEJCcIU6oVQJb73Emgt4NVfXpyeXwJi2EOr04H/VAsVnei2pUYjfbtLHiFAWxa1H/8Tm1iEGZnTBXZKMugs1YOQCEHvOGwgQOAwcu+lM4GNaMXGNn6WbYSTlAI4hQoR/Lo1AQlNJdI8zdI0GnQ69/f3NuUY22E87UiiJR05VguwlrV+DHyWILV/y7wYRjxZENDXUIFOAFef3vMJm8YMytIQsb6PvdQLpk9JIgmOzbheksbeJEtzRFM4wtBNACAbsEB7OCYX4zZ5PhxfjJ9iIz9dXH//+sdr8tPw6mp4eX1xPiavr8jp68uzi+uL15fw6zsyvPyZ/HBxefaUMA9nEsgZxTgCQNNDcgLHYFtjxnIoqCUliZjj3XoODC2YZnTKyDSExSGAEZGIxXMvwWlNAEEXm/G9uZfSlGeVxmW3AGQaDqao7JCPkxmxHNK27Q78d8cCN4w7U0Azm9gw1x2lF5d/zKjztqOqW04YpHHo+6AmYzZFgvGebWg2rz+JTb565NCUiJ7enF+NgT6P5U/2jgIhWKeuOTS7yHPoOYueZ85blg5EoyLrHCotdE7CV2fx61S0NwLKMZl1djm+Yk4Yu+K35QKZLoLbGOrFmZNClgT8KYzfslj8aCPZyAh6Q/KLFZ8FyG8JyVEzi6JQWgMyE2cJJwC6jJmTkuUQSW6Irchsfb/Q7zY1r/8pAw6FOU623gluvP/rdw/6x/v93y7SJvN/M2M+6N3ETqON9oIN9l+vD5Odn//+8dHe/ttJ+uMPi7js1gvAKsINWptY//lPq3mThvVg0eTQLbMRn06Yn4BFE9lv2UI0x39kE1i6GfCR7YUd7CrXRk0Td9TPJE5//AEWjeNnrsbUJrLiCkTKdYsIYisDUgMh++c9lUfhBcA8YBLy6vYV8xkFS+MSkKvETKPmzWG9E5gRgiXeLZnRZBRD+TvSTma0f3Q8gG7fYPfQFcLbKZ0SXSOKvSC9Je2vk//9OilCxiwKEy8N48WqJmCMrKrBwdYNwmCNcRcnxGWRHy7mLEjlDl8zR9K56+XI9aml4u+TNtH/IB5JymLYGcH/J5YfTqdg5DW7Bhv0/2GvW9j/949Ouod7/b+LZPr/bkGwg3Tipbb4C1XcXY/6Eeij1lsvcAfkVLDAd5wFWnOWUpemdAAKQChN/IsYDendU7qIQE22E9ibtgFG+AIdP8xccys1pwFoFcl5Vl1xCzeG2JPkRPwTFBiNE9i2tMTmGlT7Je8DuFRmSQCCOtyCcXOfRWyJbAkjPRlnOCgC+yIG+XOaOjNRTar+J3WYPVmN+JOWZVmtT0ryOXVmoI1rif7paVuPoaDuZ0BEJ/EsN/ZwhAaWcu4T79PTsBLBJxrBz4SESUCjZBamnxvlFF6fFcHmoevdLopmeCW9BKiiFzCA63GvmMpBGKAiN+ReIQmYUYQJrLoB+b8lWX4BOzj+xQaC/2IjP1GQzxhyH9UqcPu//ucG/7daH76vFqRC5cR7rOYw81luFNR185jf3Cgq3swzP/XApqFBeuO5NzcDEkbolwrjf2aCLXZIqRVqV4320QqY/Ky/F4Sph/9I9JLikpOXvan+504b2f9hcOtN5zSy+ObvjjnAHBa65PGcgdVuBJrO/w+PDwr2/8lR92Rv/+8iSe9Dblf/hk/sazWvwveRuyYgFzjOD69olFvcxHJV7a2pZhxZKYlolSuFZwsng1o3y84abP49ZAIvp+QQoRU6vMfkJs+lA/IeG1k56nxzhlPjU0/Zg6bt5H/trT9PDfJ/cNQryv/hcfdgL/+7SA8l2JIjLMEjeYkWPFawezW/JdrwtbmdKNksE8eD3MzHlrVAC1PXMowQrIASjc3+67uLl9fnV//WVg73xhZSbmNACLfsCiAPsttXjf3AFjdFPJa7F0JGHJ8yAnorM8ojfCX2Mje4mVF517il+WgkaN6U/3XG2rh5/msNtWq3+xmPUGzZ1h6hBjxVmzsDEIfHT3hu5mJzh/u5z3HjS8jQdYuUWLV5K+zd/mTk/TS75Q8l8qdeyrdKW9l/G94Gb7r/2Tspnv8f9I739t9O0kPZf5o3PupmTvSit3Aon+iYxX/NgXDGtRUf24apKRsoWJzScdvjDWkSSDVxapqjrcJ+Wbbn+B7gCpABbCJRCfIRAr6FfOV8ow7eK8U+oPh6EbGEk0rf7Gw3tG+XG8CLm6p+uwm/qvoSZU5klbshVkbNzdAxK2o8fos2pQrU2KxfrKD7m2Rxkm7YI6+zWZ+iSt6lgN2z1NHO2QQWFVialr5ax6dJcqkEsNAF1rRlFVtDLgcG1fGCsZcummtLQIVfmdU5+IRfPxVAJA1/xluqK+C0MB//lV0pf8q00fofu1aUTXzP8SLquniVey1LYPX63+vCal+8/9E7ON6v/7tIxXNN+NtYLt9+m4iDTWUYALXCOeziwix22Ble7OJGf85OoEEQynv4QoEZJveUBfwO3STzfFyasXFtfnTtY7uvDY0Cp4FmEct6zObMXRSWcH0QOo3DLFIWQDWoNFAGxmo/4n1djIaiL17ie0n6Q1XpSyjgEJEPVoFfgSkvhT11nF4ue7IQzotEmRdMM5/GpbpQmjghLrzaWkKEJY3k+a6gT85oSWB7NadqvYAGguHo4s3BOJeNFxCNML7CsGAJw+gMUNMYUSHD+AR65GJElMDrtkBNwPYn9fJHogYv5Q4acz1/g8jJoBxxOzDhoQJymLDwivGI+BnPQE0YYfkdGgABvuHkV1hgbTLmfocEyZ/5LvIe/EyhBSecBt7vuu0EQ2SwU67b0kKbHtgjcUB9wvevTzHEhIf3xAx7IVlgtHcn1j3yKowxUOk2HPDQnmTQwYASJUNOOJ9nIC0LEeuB4TVhnHRcdsf8TuJNLRrDBjVlPBKjA2S0OOoBlyN77n4ZS6lLvsmfQHMzDQN2gqlRwPl6xQwgZ6PdQmVVMYoloVX0xtX5+JqorvlkFKnP6W4wz3IKkGBADx6IgxFVcTjnbcLiH4VAYREewu2jQqNJNpl7qbCqYHJwrmxyyhULmTCSRaBrMNzsIoDcOfNPcX/xsScAKZ1YSNj1psDUiUVgQTWjQOmvmvkqyOoYoE15VdFSUhZM2S1QtkqSMVVLMyYvkp0WCwooKuSQqxAfCoQEAVp2qKLhSjql1C5ZcpwnR2OXgGqoTrQpnsfXWg5krfkAVZMlG8wIhy/Nicj9CLPC3sEq1DQl5xxIxdHB1NzPGA/Eq5wGTfSK+RDdYeAeiGDdXEzCEHb6RfV8Sz2fua+5qyx/t6YS5+8K4EJJ4WLMyej7skHpfOMgpSWB44QLihdzJpTOv3XYCjTAvALH1VjKwglDXCV+Yro1lhUtIo40xZv/oNRYQGiKtjZGXVLiwLRjIGmQzScwYSg6HhgEZXxXMYlIstma0sLAhhJYSXGhf56nx0TuaZXw5rrFBTcEmsRVqHMacN7BBXdqnBSYicVxGL8CDs1tiVcM4tyoIPiHLzuYC1pZZOvVCPbLxeVnmZZDxVmtZBhzFBXaSAMg+VI6j9YawbWCVvOgqystimg3cBemBrRvw3hO0wHB9dTCPrYeHgKsNTLpHMozElavw5EBD9Y1bZEXLK2y3qtAT1E7rQsM7b4R2uOVUB61kFcMo7AXa0E30LF66VIdKUGuLDSFpBJAc1B1KSBWpUSrV0ezkMYxXRTKvIoR5I2FM8UDnltrF3yQFSC3kavR4H4ypenw73VQqSDT5shxJwtudqAIbYcmTEdFeIW22RC3NwqDqMD2gShcZ2cJW6HZyCplikck9O3spRvUyMkmeh+0dJZyU438sfcnbp429f9Ji0oaVOsdBDb4/3q942L81+HJyf78byfp8/X/5TntY3r/KlbupfMvX1jy/RWwrPH83c0Lbr98tV16/fLjqXT6FXZNe5ff3uWXp/7e5bcDl19eUNfy+EnBLVC2JMaY6p0GszBJ1zCdv5dgyg5V1ZT1+YO+5UWC0FWPOpV4CVOtvmmgNiZgD9h/uYvX8Y8goNAAvk3UZPVX1cFxmO6xAv4VaKsRlbxMkCUfvELmjbEr9Jxly84286Apg+zibJ1NAgfM7Q7wKSzIqp6YDQb2AHsDxSWF7MpZLMAsqfAg/tuCeK3rvn0QEftw522dY3Pvut27bveuW572rtt1h7d33e5dtzpt5brdQll/HMftwyKyAydttT2B6SGGsHfR7tPqtIn/d/mA22YBIA3+3/7BYfH9z4ODk/37/ztJed+veIlPeEXP9GyvHQXSWi/2g7+FaD46MA3CmCmvsryRX/Iia6VSjN4Q1QekjSqonb9Wv0mQCXYy86Yzi96B5UUnnu+lCxnYYtf2LnUv939pP3TMhLr/3kP9uHiJL1MPSI+X8Ae7k9xNfJl5GmZBKjBJgG74HIEYOg+fe7neUI5lA/i8MpvKt23Q+w5rw4/ciaaU8py++zGQY/XzgQX5omUkAeSPs3haAuaZAkypC4l5wSFWmlVMa75BocCdGXPeJtnceI9AhPhVPi6R49VH/JVL8pV9LbG0nwN3jmg6I+21njdpP+a0Fi90Ag4mXoV4jhpUV8ZPbYFsA1prCsC3qobplIT+Qxj54nQZfKKXi2SRwExb/+h21RIPEuA5bOg4yMWXq7UEp40O+FT9WU3KRSQ+O3ko+aBqGWyU+f4oBOHKh76Ix0sjXZibs3A+p6Yn3SKd5vdwl8CWBSoPC7gII6jSHZYfOm+fGWgIwHMJp3XMSwAzMcI2RSTNhL8Bb4HEIcM4WRzD5Fl4MBE4ns+SZ/nwHnnqlthmbXtZc7wInKS6J4ZPy2/bEa/c3A9/sH6bPhIVNLaqfSEhFn7H4VmHpU6nmq2kJHWM0MJiM9hzhC/pb46rWbsR4QSWianYVW/Rk1G7qSc3SGL+KYDNu9FVm/qYMeqnM64AN+/FqLxGP3E6YTS1tOnxrM7yIDU1oXF2b/EDwDvqWwni565CTtSzeb0LWW0sapW68XLfWNicFvn6TeQQxpCl3TPWcsWt7YBX0Q7Roa5QbPuefxhi8xGIek2Y37PJLAzfKrltns1VteVqZOFHKUy0JJiI6bUl1Ai/XNHcHr4s0Ngci6tac70EDSnj8YEcsWTxMuY4gZX51xA2/O2n7bq2ZN9VDf0ki2paKdhbYKPFnpNU4S2LrAmYCZa8M/ZsYNp+1XUrTKGlTqhvTZRXIaIXYHlfwBy0NshlWcVQ2W9laHxeU8eIk7b43EmxXwTSS31lp/lmqqrj105qMOaVRXkD+VhwZ1okwlB6eT48O7+6OX95foof3bm5HL46H4+Gp+eGZ4cf8n8XhwU/7K3HfPeK3RY9WjwfbcyBtt1tLYrbWuwK34tXwxfnbwDZ11c3r9+cX/10dXFdwnVAOvxLMMabbZ3KR9xWkcv37oDASTKKw0nOa4e3BF6wND/siI+3I9jv93wRMGPOdKzmUUz8kgmg//319cgo4FekqH/GfLqQawRsBbsaAs+HvY1x5afKu0G1v8QVu0jKfKiUpdwFLxvUNv6oiNs6mpM7YNPQCf0BuT4dFQPS47zDQFFTux7LoenLGu9JIHc+vW5FLD6mu9DP5uwV7mYqhqzfOVNpjoBCcJqtzQ8VpLpnFKuQKQmTAYdc9DrwF4b7VmGVowZ+/g3v0QwMTnCHQeoNSwVLp/NZhv5ovIzmZuh9uOB2hsw+f8eczHwSQwyPb03HOeeHMSp0g5yLb4hVHUfzt3prP+WhP/ZRcp+rZ3xAQZVP48RVr7Lj31rnkyFmhTSMQj+cLn5AHNv5D4moaxFtWUPwXmljXGAeRz3TYmK39istKrnslmZ++ip0od6hFvaNOHM9vtwc3yY+X4H7/oWJtdIm/n+QeTAw44yfdk0yd8rWOwhoev/3uPT936OD3v77TztJUswTNr/Dp47mEX6Vs/3fz0jP7vesLsjhKY2EL9wDNYCXx+R5gf3CS98sLW7zICHiTjX9UR/5naFKiAkYmT1z+ZUPMoTumWa155zVHvoUYrPDAeUXrXCaJ9mEPEJPbZU7/zHpPYhH/2PN/ybyH0+os82HwBvkH/R26ftvvf4+/mMnqRj/weeYZuksjL3fhTOxFAQivmtxFfpsE5ncRNrkhwssjC14gTEdMgai9tHgVs7yR1DT5Z2UczriSoJRgE5rj1Xk5EG57zn/wwSQ38iSPwzvb0WOWU/7VYs/TaC8K7AyzwQXfrfc38tiUPcTSSpUrtz4FsEtAIzaqSU+vaL+Ehfwy1NSdzBanhHx3Karc/NItJ+0y407IZDAC0w+LLfLdXuhNQd2NtxSFKMiy2GRD+1F5esAH6vwxd/i+Y+gcVnPaj9y9WzUEbzd5v/gYb3kQjVFtbIrKrjyg8m5r++aABWhX7wAtinIPvjgex119IKbFH52YH9Ife93xYcMwyvEnwmDWUq1XIiNiYSS4SJSGPRNBFWRu21zP6g4d8wJIQgaK2Wg8xFGIvKXEKWiX8OJ+AOs3uUfHfH6MTBzlvJPHEs/hmM+Ein75LFyik6ujpVLTJpyO8hbSxbUHfOERlzAKucBazY3BTwSJg71V80ohlCBEePDqBW4mMPGxlcG322Fr7AWy7UrNiT1emATLSe4zmcofx+0QD4XLPXR1knoQjpxFclWYNjSj5saK3gDPmDc4uU/vhiLyuPcQf/D2OKf2gr6+6ZN7H+hsNXZGt5OwYfr/MYtQeP+vxT/fQybgr39v4uEO++YBlOmd6/qUFRNL3evF7Wg1nVjzhMV+k2fwKp28OaTuNFkizfUL86IfB13+ZdoLneV6T36pgOH9PpV+uSrps097rbr34MuXOaLsjgKE7zNzDG0yoMQMqAfpFYjEbpNj4tjoMYmyvRIjTIxWrNcjh9gxL3C1xH9LWMf8cnczeSfU2NjF0CT/J/0Tsrff99//2snqUao+UT/Te79KteekDhpG12Mcj679S7d6uNRq3wY23jzJH/WyU+MaAxG60ZHpxvP/xbyL7da66uBJvk/6B4V33/p94/28r+LtEr+lYn/SR3vuPnkx9l5pK7Dtww/wwv2CdtvH7ZPG8m/iPXiXxjYxAhoXP+PD4vxPye9vf9/J2mbC3KmzlAfjij6QMYin4cOVCuQDT5isb5NIJw2FnCxF9xR33Mt4dHRVsKGysf3w/s3/DLG+buIBmLM/K6MDunED483DMeAxVbF7Q7pGhJ3FxoaKNXAZiIaA4XUZ78lqFychS0zitncy+Y3L6/GdVuITeT/LqLbHP81yX/voPT93353//3P3aTCNR+cYtjETnzmFgS9XXYag7S3pbi/kd7iUegOtbd4/a9HQbdrGg/KWK/AWykFM6Qnn7d8jk5eShTOd/VKmy4Q0UvfPFFvPs29YIiawAykdqKs5ltBZVzsZQM21Kv+XlBzNeNC2ZzNw3ixFQKi6jY4yJocjdxuS2xU9G3iqphSzC/FlTbd/wIAob7N6RM5JbWJqJvA9hJu/9Gh+tSs/+Vbd9sofpka9P/B8WHR/9s96u39PztJIpiSa8ooTLyUaxWW2VOHP86p2QOYAs+BdEZn6UTtlEIhUzodLB9ZjIwYzIvbyzAdiTe2Wi3zohSGRxcvV/EcGXB81P0aK6jzP3xKAkQeMZfrlb7DXKXzyxqc6++j7ryVV6jHh6+8RrXTxjWu3WoZ0S8D8m33225reeWeZ/RaLSPkBxsyL6Ho5cgMiwKkNBiPo1wJlSyvUFdCmPdNVoEZwYMrwPStlBUwRujeaihx92FApEFfjqQbkIMuluZvuKxoVNxxWQFQG+mmHAmtciTWgPzr361CXBXPK3gvdRNfkqq70vi5yi+JjEoGTvjSuDUd0SwRsWHc+uBlYNDzdq8MsZx66SybwAZh3lleHzf/nPjhpDOn6L/s8Bd2O7zpzlkI7BZjJKxs2xR2Jelh+P/tnckOwiAQhu88hem92IHqoQ+gj+DRGLfUPdaDjy/LIMVi4tXk/25lC4RCOwwzsz9tl9Eq29ctV+fNtOZqbl0XWlYFJ7x995Ikks//HhUNRuXvwZLyGVJKIZLDVyd3xZPcZlTXmpOCyQtValIJkVp6N71dyhp7N+F2kVi/282HDs0FDrUurMwuaguNDx07FOtF1PxSwoXXJLbf59iXpO1jjET5EYdyPXQE4OXuchc9scVYk1rPW05MpOXU9sHfmTIr7MhlM+LxYtU+Ztf7rDW9NIPvTP5diNCHRvTnrHAzEDb68G1ip9C82exDUTOxA32nXZxlT7PHukByL0FU6nW3S5IWlHmsJad8KyrTivreisLfKgAAAAAAAAAAAAAAAAAAAAAAgF94Af9NgUUAyAAA
```

**Review by MartinWeindel — APPROVED** [review]
/lgtm

**Review by MartinWeindel — APPROVED** [review]
/lgtm

**Comment by timuthy** [comment]
Nice, thanks for the fix 🙂

**Pull request merged**